util/src/main/scala/com/evolutiongaming/util/BlockingTracker.scala

```diff
@@ -0,0 +1,42 @@
+package com.evolutiongaming.util
+
+import com.evolutiongaming.util.ExecutionThreadTracker.stackTraceToString
+import com.typesafe.scalalogging.LazyLogging
+
+import scala.concurrent.BlockContext.withBlockContext
+import scala.concurrent.{BlockContext, CanAwait}
+
+object BlockingTracker extends LazyLogging {
+
+  trait Surround {def apply[T](f: () => T): T }
+
+  lazy val Empty = new Surround {
+    override def apply[T](f: () => T): T = f()
+  }
+
+  def apply(enabled: Boolean): Surround = {
+    if (enabled) {
+      new Surround {
+        override def apply[T](f: () => T): T = {
+
+          val nonTracking = BlockContext.current
+
+          val tracking = new BlockContext {
```

Comment:
how is it supposed to work ?

Reply:
Tracking block context is going to intercept `Await.result` or `Await.ready` since they both use `scala.concurrent.blocking`, which under the hood uses `BlockContext#blockOn`. One may use tools such as Intellj Idea to find calls `Await`, therefore the difference is only that we will have a proof of blocking call on the given dispatcher.  And there is a disadvantage that it will always log stack trace upon event thus may cause startup slowdown :( . Intended as an experimental ground :)